test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/ProjectDirectory.cs

```diff
@@ -78,12 +88,22 @@ void CreateDirectoryProps(string originalProjectRoot, string projectRoot)
 #else
 #error Unknown Configuration
 #endif
+
+                var references = GetBinaryReferences();
+                var referencesItems = string.Join(Environment.NewLine, references.Select(path => 
+                {
+                    var name = Path.GetFileNameWithoutExtension(path);
+                    return $@"    <Reference Include=""{name}"" HintPath=""{path}"" />";
+                }));
                 var text = $@"
 <Project>
   <PropertyGroup>
     <OriginalProjectRoot>{originalProjectRoot}</OriginalProjectRoot>
     <_RazorMSBuildRoot>$(OriginalProjectRoot)\..\..\..\src\Microsoft.AspNetCore.Razor.Design\bin\{configuration}\netstandard2.0\</_RazorMSBuildRoot>
   </PropertyGroup>
+  <ItemGroup>
+{referencesItems}
```

Comment:
@rynowak  - wip

Reply:
Btw, I get a msbuild unification warning about System.Diagnostics.DiagnosticSource. I didn't get a chance to dig in to it last night, but we do some special casing around the library in all other places where we use the binary that I might be missing